pkg/rke2/rke2.go

```diff
@@ -166,5 +176,145 @@ func setup(clx *cli.Context, cfg Config) error {
 	}
 	executor.Set(&sp)
 
+	disabledItems := map[string]bool{
+		"kube-apiserver":          disableAPIServer,
+		"kube-scheduler":          disableScheduler,
+		"kube-controller-manager": disableControllerManager,
+		"etcd":                    disableETCD,
+	}
+	return removeOldPodManifests(dataDir, disabledItems)
+}
+
+func podManifestsDir(dataDir string) string {
+	return filepath.Join(dataDir, "agent", defaultPodManifestPath)
+}
+
+func removeOldPodManifests(dataDir string, disabledItems map[string]bool) error {
+	var (
+		kubeletStandAlone         bool
+		containerdErr, kubeletErr chan error
+	)
+	ctx := context.Background()
+
+	for component, disabled := range disabledItems {
+		if disabled {
+			manifestName := filepath.Join(podManifestsDir(dataDir), component+".yaml")
+			if _, err := os.Stat(manifestName); err == nil {
+				kubeletStandAlone = true
+				if err := os.Remove(manifestName); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	if kubeletStandAlone {
+		kubeletCmd := exec.Command("kubelet")
+		containerdCmd := exec.Command("containerd")
+
+		tmpAddress := filepath.Join(os.TempDir(), "containerd.sock")
+
+		// start containerd
+		go startContainerd(ctx, dataDir, containerdErr, tmpAddress, containerdCmd)
+		// start kubelet
+		go startKubelet(ctx, dataDir, kubeletErr, tmpAddress, kubeletCmd)
+
+		// check for any running containers from the disabled items list
+		go checkForRunningContainers(ctx, tmpAddress, disabledItems, kubeletCmd, containerdCmd)
+
+		go func() {
+			logrus.Infof("containerd 4 %v", containerdCmd)
+			select {
+			case err := <-kubeletErr:
+				logrus.Infof("kubelet Exited: %v, exiting Containerd", err)
+				// exits containerd if kubelet exits
+				containerdCmd.Process.Kill()
+			case err := <-containerdErr:
+				logrus.Infof("Containerd Exited: %v, exiting kubelet", err)
+				kubeletCmd.Process.Kill()
+			case <-time.After(5 * time.Minute):
+				// kill both temp process after 5 minutes anyway
+				kubeletCmd.Process.Kill()
+				containerdCmd.Process.Kill()
+			}
+		}()
+	}
+
 	return nil
 }
+
+func startKubelet(ctx context.Context, dataDir string, errChan chan error, tmpAddress string, cmd *exec.Cmd) {
+	args := []string{
+		"--fail-swap-on=false",
+		"--container-runtime=remote",
+		"--containerd=" + tmpAddress,
+		"--container-runtime-endpoint=unix://" + tmpAddress,
+		"--pod-manifest-path=" + podManifestsDir(dataDir),
+	}
+	cmd.Args = append(cmd.Args, args...)
+	cmd.Stdout = nil
+	cmd.Stderr = nil
+
+	errChan <- cmd.Run()
+}
+
+func startContainerd(ctx context.Context, dataDir string, errChan chan error, tmpAddress string, cmd *exec.Cmd) {
+	// starting containerd first then kubelet
```

Comment:
does this comment belong here?

Reply:
yeah that was before the code cleanup, deleting it now